packages/effection/src/mailbox.ts

```diff
@@ -52,23 +57,45 @@ export class Mailbox {
     };
   }
 
-  *subscribe(
-    emitter: EventEmitter | EventTarget,
-    events: string | string[],
-    prepare: (event: { event: string; args: unknown[] }) => unknown = x => x
-  ): Operation {
-    for (let name of [].concat(events)) {
-      let listener = (...args) => {
-        this.send(prepare({ event: name, args }));
+  *pipe(other: Mailbox<T>) {
+    let that = this; // eslint-disable-line @typescript-eslint/no-this-alias
```

Comment:
Typescript doesn't like this pattern, but I'm not sure how else to get around the scope change introduced by `function*`.